fit_multiband.py

```diff
@@ -0,0 +1,255 @@
+        doc="Output of the schema used in deblending task",
+        name="{name_output_coadd}Coadd_{name_output_cat}_schema",
+        storageClass="SourceCatalog"
+    )
+
+    def adjustQuantum(self, datasetRefMap):
+        """ Validates the `lsst.daf.butler.DatasetRef` bands against the
+        subtask's list of bands to fit and drops unnecessary bands.
+
+        Parameters
+        ----------
+        datasetRefMap : `NamedKeyDict`
+            Mapping from dataset type to a `set` of
+            `lsst.daf.butler.DatasetRef` objects
+
+        Returns
+        -------
+        datasetRefMap : `NamedKeyDict`
+            Modified mapping of input with possibly adjusted
+            `lsst.daf.butler.DatasetRef` objects.
+
+        Raises
+        ------
+        ValueError
+            Raised if any of the per-band datasets have an inconsistent band
+            set, or if the band set to fit is not a subset of the data bands.
+
+        """
+        datasetRefMap = super().adjustQuantum(datasetRefMap)
+        # Check which bands are going to be fit
+        bands_fit, bands_read = self.config.get_band_sets()
+        bands_needed = bands_fit.union(bands_read)
+
+        bands_data = None
+        has_extra_bands = False
+
+        for type_d, ref_d in datasetRefMap.items():
+            # Datasets without bands in their dimensions should be fine
+            if 'band' in type_d.dimensions:
+                bands = [dref.dataId['band'] for dref in ref_d]
+                bands_set = set(bands)
+                if bands_data is None:
+                    bands_data = bands_set
+                    if bands_data != bands_fit:
+                        if not bands_needed.issubset(bands_data):
+                            raise ValueError(f'bands_needed={bands_needed}) !subsetof bands_set={bands_set}'
+                                             f' from refs={ref_d})')
+                        has_extra_bands = True
+                elif bands_set != bands_data:
+                    raise ValueError(f'Datatype={type_d} bands_set={bands_set}) != previous={bands_data})')
+                if has_extra_bands:
+                    for dref in ref_d:
+                        if dref.dataId['band'] not in bands_needed:
+                            ref_d.remove(dref)
+        return datasetRefMap
+
+
+class MultibandFitSubTask(pipeBase.Task):
+    ConfigClass = pexConfig.Config
+
+    def __init__(self, schema: afwTable.Schema, **kwargs):
+        """ Initialize the task, defining and setting the output catalog schema.
+
+        Parameters
+        ----------
+        schema : `lsst.afw.table.Schema`
+            The input schema for the reference source catalog, used to initialize the output schema.
+        kwargs
+            Additional arguments to be passed to the subclass constructor(s).
+
+        Notes
+        -----
+        This method must initialize an attribute named schema with the schema of the output catalog.
```

Comment:
If you can make this an ABC, you can do
```
@property
@abstractmethod
def schema(self) -> afwTable.Schema:
    raise NotImplementedError()
```
if you don't mind forcing subclasses to use a property - it's a bit more pain for subclasses, but you get some nice declarative clarity.  Your call.